### ISSUE TITLE
Add METADATA check for Han Unification

### ISSUE THREAD

**Issue by davelab6** [googlefonts/fontbakery#2506]
If a family supports one of the 4 CJK scripts, it should not support any of the other 3.

**Comment by felipesanches** [comment]
Did it actually happen already?

**Comment by davelab6** [comment]
Did what happen? We have some Chinese, Japanese and Korean fonts already

**Comment by felipesanches** [comment]
> Did what happen? We have some Chinese, Japanese and Korean fonts already

is there any family supporting more than one of the 4 CJK scripts?

**Comment by davelab6** [comment]
Noto Sans HK and some of the Chinese Simplified fonts were detected as
supporting several. The metadata generator should be fixed too, but this is
a fail safe request

**Comment by felipesanches** [comment]
The git repo at https://github.com/googlefonts/noto-cjk contains ttc files, but I don't see them at https://github.com/google/fonts/

In order to work in this issue I'm interested in looking at the bad METADATA.pb files that the check must detect in a FAIL scenario.

**Comment by felipesanches** [comment]
@m4rc1e do you have a script that will sync fonts in the API to the git repo?

**Comment by felipesanches** [comment]
ping, @m4rc1e

**Comment by m4rc1e** [comment]
> do you have a script that will sync fonts in the API to the git repo?

@felipesanches apologies but I don't fully understand. I'm assuming you want a script that will take a family from fonts.google.com and generate us a family dir in our google/fonts repo?

The only script I have is a reporter that will tell me which families haven't made it into fonts.google.com yet.

**Comment by felipesanches** [comment]
A few weeks ago I was trying to work on this issue and I was unable to find where the CJK fonts on the google/fonts repo. Then @davelab6 suggested that perhaps the CJK fonts were pushed to the API directly (without going through the google/fonts repo first) and that perhaps you had a script that could be useful in this case. I simply copied here Dave's original question to you, so I am unsure exactly what he meant.

**Comment by felipesanches** [comment]
let's make things simpler here: what are the valid values for the `subsets` field in a METADATA.pb of CJK fonts?

**Comment by felipesanches** [comment]
@chrissimpkins, maybe you can help us clarify here what to do in this check (and how to do it). Dave's original description for this issue is too short and lacks a more detailed definition of the problem.

**Comment by felipesanches** [comment]
@chrissimpkins agreed to be assigned this issue so that he'll be working on this new check request to be part of our new CJK fontbakery profile.

**Comment by chrissimpkins** [comment]
Sure happy to help here.  Please let me know where I can find the CJK fonts and metadata files that you would like to have reviewed.  I don't have any specific FB check in mind at the moment because I don't understand what process is failing here.  Please add detail so that I can understand.

**Comment by felipesanches** [comment]
we need a more detailed description of the issue from @davelab6

**Comment by felipesanches** [comment]
@chrissimpkins I've pushed a commit to my personal account with a boilerplate of a new CJK profile. Please use that to prepare a PR when you start creating CJK checks. A single cjk is enough for me to review and merge the PR and then you can gradually add more checks as usual.

I hope this is useful. The commit is available at https://github.com/felipesanches/fontbakery/commit/9fde1ce000ed4a88a254f6aca00c72d0a45850f1

**Comment by chrissimpkins** [comment]
Thanks Felipe! I will have a look.  I really appreciate it

**Comment by chrissimpkins** [comment]
Here is the list of fonts that display by locale:

### Simplified Chinese

- https://fonts.google.com/?subset=chinese-simplified

### Traditional Chinese - HK

- https://fonts.google.com/?subset=chinese-hongkong

### Traditional Chinese - not otherwise specified

- https://fonts.google.com/?subset=chinese-traditional

### Japanese

- https://fonts.google.com/?subset=japanese

### Korean

- https://fonts.google.com/?subset=korean

**Comment by chrissimpkins** [comment]
#### Simplified Chinese

1. Noto Sans SC - no METADATA.pb file available in [source repository](https://github.com/googlefonts/noto-cjk)
2. Noto Serif SC - no METADATA.pb file available in [source repository](https://github.com/googlefonts/noto-cjk)
3. ZCOOL QingKe HuangYou - no METADATA.pb file available in [source repository](https://github.com/googlefonts/zcool-qingke-huangyou)
4. ZCOOL KuaiLe - no METADATA.PB file available in [source repository](https://github.com/googlefonts/zcool-kuaile)
5. ZCOOL XiaoWei - no METADATA.pb file available in [source repository](https://github.com/googlefonts/zcool-xiaowei)

**Comment by chrissimpkins** [comment]
#### Traditional Chinese - Hong Kong

1. Noto Sans HK - no METADATA.pb file available in [source repository](https://github.com/googlefonts/noto-cjk)

**Comment by chrissimpkins** [comment]
#### Traditional Chinese - NOS

1. Noto Sans TC - no METADATA.pb file available in [source repository](https://github.com/googlefonts/noto-cjk)
2. Noto Serif TC - no METADATA.pb file available in [source repository](https://github.com/googlefonts/noto-cjk)

**Comment by chrissimpkins** [comment]
#### Japanese

1. Noto Sans JP - no METADATA.pb file in source repository
2. Noto Serif JP - no METADATA.pb file in source repository
3. M PLUS 1p - no METADATA.pb file in the [source repository](https://github.com/rayshan/mplus-fonts)
4. Sawarabi Mincho - unable to locate source repository
5. M PLUS Rounded 1c - no METADATA.pb file in the [source repository](https://github.com/rayshan/mplus-fonts)
6. Kosugi - unable to identify the source repository
7. Kosugi Maru - unable to identify the source repository
4.

**Comment by chrissimpkins** [comment]
Seems like a consistent pattern across the CJK fonts.  There are no METADATA.pb files for any of these CJK projects as far as I can tell. Removing my assignment for this issue as I don't understand what is being requested here.  If someone can expand on the needs for this check, I'd be happy to revisit this issue.

**Comment by felipesanches** [comment]
@davelab6 could you please clarify the requirements here?

**Comment by davelab6** [comment]
I see that the METADATA files made internally haven't been reposted upstream. I will try to get that done

**Comment by davelab6** [comment]
This is literally only about the METADATA files, not about font data.

**Comment by felipesanches** [comment]
Once we get the METADATA files available on their respective repos, I will reassign this issue to myself and use the METADATA files as a reference for implementing (and testing) this rather simple check.

**Comment by felipesanches** [comment]
All of Font Bakery's `"new check proposal"` issues are being either closed or migrated to the [Fontspector issue tracker](https://github.com/fonttools/fontspector/issues), as there's no intention to implement new checks on the Font Bakery codebase anymore.

If you believe this proposal is still valid, please post a message here asking for it to be migrated. Otherwise, on **July 20, 2025** (30 days from now) it will be closed due to inactivity.
